core/domain/state_domain_test.py

```diff
@@ -1138,6 +1138,8 @@ def test_interaction_validation_for_android(self):
 
         init_state = exploration.states[exploration.init_state_name]
         # Valid interactions.
+        init_state.update_interaction_id('AlgebraicExpressionInput')
```

Comment:
It's a bit concerning to me that these were missed before (i.e. that we weren't verifying whether they were valid on Android). How hard would it be to update these tests to be generic, e.g. something like (pseudocode):

```python
all_interaction_ids = fetch_all_interaction_ids()
android_interaction_ids = android_validation_constants.VALID_INTERACTION_IDS
web_only_interaction_ids = set(all_interaction_ids) - set(android_interaction_ids)
# TODO: Verify len(android_interaction_ids) != 0 && len(web_only_interaction_ids) != 0
for android_interaction_id in android_interaction_ids:
    init_state.update_interaction_id(android_interaction_id)
    # TODO: Add message with interaction ID to simplify debugging.
    self.assertTrue(init_state.interaction.is_supported_on_android_app())
for web_only_interaction_id in web_only_interaction_ids:
    init_state.update_interaction_id(web_only_interaction_id)
    # TODO: Add message with interaction ID to simplify debugging.
    self.assertFalse(init_state.interaction.is_supported_on_android_app())
```

I'm generally not keen on generic tests like this, but it seems necessary to ensure we don't miss any interactions.

Alternatively, a slightly crisper way to do this might be:

```python
_checked_interaction_ids = set()

def _create_init_state_for_interaction_verification(self):
    exploration = exp_domain.Exploration.create_default_exploration('0')
    return exploration.states[exploration.init_state_name]

def _verify_interaction_supports_android(self, interaction_id):
    init_state = self._create_init_state_for_interaction_verification()
    init_state.update_interaction_id(interaction_id)
    self.assertTrue(init_state.interaction.is_supported_on_android_app())
    self._checked_interaction_ids.add(interaction_id)

def _verify_interaction_does_not_support_android(self, interaction_id):
    init_state = self._create_init_state_for_interaction_verification()
    init_state.update_interaction_id(interaction_id)
    self.assertFalse(init_state.interaction.is_supported_on_android_app())
    self._checked_interaction_ids.add(interaction_id)

def _verify_all_interaction_ids_checked(self):
    all_interaction_ids = fetch_all_interaction_ids()
    missing_interaction_ids = set(all_interaction_ids) - self._checked_interaction_ids
    # TODO: Add assertion message with the results of missing_interaction_ids to help developer know which interaction IDs are missing.
    self.assertEqual(len(missing_interaction_ids) == 0)

...

def test_interaction_validation_for_android(self):
    self._verify_interaction_supports_android('AlgebraicExpressionInput')
    ...
    self._verify_interaction_does_not_support_android('CodeRepl')
    ...
    self._verify_all_interaction_ids_checked()
```

WDYT? I slightly lean toward solution 2 since it's more explicit.

Reply:
@aks681 do you have any thoughts on this as a code owner of state_domain_test?

Reply:
Yep, I think 2nd solution is better. I've made the changes. PTAL!